lib/cache-utils.js

```diff
@@ -55,18 +55,18 @@ function wellKnownCachePath() {
 
 export async function getCacheKey() {
   const platform = process.env.RUNNER_OS;
-  const fileHash = await glob.hashFiles(await getScarbManifestPath());
+  const fileHash = await glob.hashFiles(await getScarbLockfilePath());
 
   if (!fileHash) {
     throw new Error(
-      "failed to cache dependencies: unable to hash Scarb.toml file",
+      "failed to cache dependencies: unable to hash Scarb.lock file",
     );
   }
 
   return `scarb-cache-${platform}-${fileHash}`.toLowerCase();
 }
 
-async function getScarbManifestPath() {
+async function getScarbLockfilePath() {
   const { stdout, exitCode } = await exec.getExecOutput("scarb manifest-path");
```

Comment:
I don't think this is logically correct :/ 

`scarb manifest-path` returns *package* manifest, and the lockfile is produced besides the *workspace* manifest. 
E.g. in scarb `examples/workspaces`, when run in `addition` package the `manifest-path` will return `/Users/(..)/scarb/examples/workspaces/crates/addition/Scarb.toml`. 

I wonder if it would make more sens to traverse the directory tree upwards looking for `Scarb.lock` file directly 🤔